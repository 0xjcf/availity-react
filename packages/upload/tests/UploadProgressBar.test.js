import React from 'react';
import { render } from '@testing-library/react';
import MockUpload from './mockUpload';
import { UploadProgressBar } from '..';

const instance = new MockUpload();

describe('UploadProgressBar', () => {
  afterEach(() => {
    instance.reset();
  });
  test('should render', () => {
    render(<UploadProgressBar upload={instance} />);
  });

  test('should render progress', () => {
    const { getByTestId } = render(<UploadProgressBar upload={instance} />);
    instance.progress(50);

    getByTestId('upload-progress');
  });

  test('should render success', () => {
    const { getByTestId } = render(<UploadProgressBar upload={instance} />);
    instance.success();

    const progressBar = getByTestId('upload-progress');

    expect(progressBar.className).toContain('progress-complete');
  });

  test('should render error', () => {
    const { getByTestId } = render(<UploadProgressBar upload={instance} />);
    instance.error('File upload rejected');

    getByTestId('upload-error-message');
  });

  test('should render password verification button', () => {
    const { getByTestId } = render(<UploadProgressBar upload={instance} />);
    instance.error('Encrypted files require a password', 'encrypted');

    getByTestId('password-form-encrypted');
  });

  // 1 Test for striped and animated
  // striped
  // animated


});
