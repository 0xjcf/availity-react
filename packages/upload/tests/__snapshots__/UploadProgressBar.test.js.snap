// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProgressBar should render 1`] = `
<div
  className=" undefined progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin="0"
    aria-valuenow={0}
    className="progress-bar bg-success"
    role="progressbar"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    <span
      className="sr-only"
    >
      0
      % Complete
    </span>
  </div>
</div>
`;

exports[`UploadProgressBar should render error 1`] = `
<span
  className="text-danger"
>
  File upload rejected
</span>
`;

exports[`UploadProgressBar should render password verification button 1`] = `
Array [
  <span
    className="text-danger"
  >
    Encrypted files require a password
  </span>,
  <div
    className="pwRequired"
  >
    <button
      aria-label={null}
      className="btn btn-primary btn-sm"
      onClick={[Function]}
      type="button"
    >
      Enter password
    </button>
  </div>,
]
`;

exports[`UploadProgressBar should render progress 1`] = `
<div
  className=" undefined progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin="0"
    aria-valuenow={50}
    className="progress-bar bg-success"
    role="progressbar"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <span
      className="sr-only"
    >
      50
      % Complete
    </span>
  </div>
</div>
`;

exports[`UploadProgressBar should render success 1`] = `
<div
  className=" progress-complete progress"
>
  <div
    aria-valuemax={100}
    aria-valuemin="0"
    aria-valuenow={100}
    className="progress-bar bg-success"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <span
      className="sr-only"
    >
      100
      % Complete
    </span>
  </div>
</div>
`;
